### PR TITLE
Fix histogram chart calculation

### LINKED ISSUE #1967: Histogram metrics chart value at the end is wrong
The histogram chart appears to keep reporting the highest value at the end of the time period.

For example, it keeps reporting 0.25 seconds at the end of this chart:

metrics-histogram-chart-2

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspire/issues/1967
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1968)

### CODE REVIEW

File: src/Aspire.Dashboard/Components/Controls/ChartContainer.razor.cs [hunk 1]
```diff
@@ -170,13 +170,18 @@ protected override async Task OnParametersSetAsync()
 
     private OtlpInstrument? GetInstrument()
     {
+        var endDate = DateTime.UtcNow;
+        // Get more data than is being displayed. Histogram graph uses some historical data to calculate the bucket ranges.
+        // It's ok to get more data than is needed here. An additional date filter applied when building chart values.
```
Comment (drewnoakes) [inline src/Aspire.Dashboard/Components/Controls/ChartContainer.razor.cs:175]: The last sentence feels like something's missing. Should it be:

```suggestion
        // It's ok to get more data than is needed here. An additional date filter is applied when building chart values.
```

**Review by drewnoakes — APPROVED** [review]

File: src/Aspire.Dashboard/Components/Controls/ChartContainer.razor.cs [hunk 2]
```diff
@@ -170,13 +170,18 @@ protected override async Task OnParametersSetAsync()
 
     private OtlpInstrument? GetInstrument()
     {
+        var endDate = DateTime.UtcNow;
+        // Get more data than is being displayed. Histogram graph uses some historical data to calculate the bucket ranges.
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ChartContainer.razor.cs:174]:
```suggestion
        // Get more data than is being displayed. Histogram graph uses some historical data to calculate bucket counts.
```

**Comment by davidfowl** [comment]
If you want to backport to p3, I think its fine @JamesNK. We have some other changes going in as well.

**Pull request merged**